src/py/mat3ra/made/tools/build/defect/builders.py

```diff
@@ -371,3 +378,143 @@ def _generate(self, configuration: _ConfigurationType) -> List[_GeneratedItemTyp
+        cart_basis.cell.vector1 = vector_a_prime
+        cart_basis.cell.vector2 = vector_b_prime
+        material.basis = cart_basis
+
+        new_lattice = material.lattice.clone()
+        new_lattice.a = new_lattice.a + delta_a_cart
+        new_lattice.b = new_lattice.b + delta_b_cart
+        material.lattice = new_lattice
+        return material
+
+    def _update_material_name(self, material, configuration):
+        updated_material = super()._update_material_name(material, configuration)
+        new_name = f"{updated_material.name}, {configuration.steps_number}-step Terrace {configuration.cut_direction}"
+        updated_material.name = new_name
+        return updated_material
+
+    def _rotate_material(self, material: Material, axis: List[int], angle: float) -> Material:
+        """
+        Rotate the material around a given axis by a specified angle.
+
+        Args:
+            material (Material): The material to rotate.
+            axis (List[int]): The axis to rotate around, expressed as [x, y, z].
+            angle (float): The angle of rotation in degrees.
+        Returns:
+            Atoms: The rotated material.
+        """
+        if material.basis.is_in_cartesian_units:
+            crystal_basis = material.basis.copy()
+            crystal_basis.to_crystal()
+            material.basis = crystal_basis
+        atoms = to_ase(material)
+        atoms.rotate(v=axis, a=angle, center="COM")
+        atoms.wrap()
+
+        return Material(from_ase(atoms))
+
+    def create_terrace(
+        self,
+        material: Material,
+        cut_direction: Optional[List[int]] = None,
+        pivot_coordinate: Optional[List[float]] = None,
+        steps_number: int = 1,
+        use_cartesian_coordinates: bool = False,
+        rotate_to_match_pbc: bool = True,
+    ) -> List[Material]:
+        """
+        Create a terrace at the specified position on the surface of the material.
+
+        Args:
+            material: The material to add the terrace to.
+            cut_direction: The direction of the cut in lattice directions.
+            pivot_coordinate: The center position of the terrace.
+            steps_number: The number of steps to bunch.
+            use_cartesian_coordinates: Whether to use Cartesian coordinates for the center position.
+            rotate_to_match_pbc: Whether to rotate the material to match the periodic boundary conditions.
+        Returns:
+            The material with the terrace added.
+        """
+        if cut_direction is None:
+            cut_direction = [0, 0, 1]
+        if pivot_coordinate is None:
+            pivot_coordinate = [0.5, 0.5, 0.5]
+
+        new_material = material.clone()
+        material_with_additional_layers = self.create_material_with_additional_layers(new_material, steps_number)
+
+        normalized_direction_vector = self._calculate_cut_direction_vector(material, cut_direction)
+        condition = CoordinateConditionBuilder().plane(
```

Comment:
Why do we need to initiate this Builder? Should be `CoordinateConditionHolder` and `CoordinateConditionHolder.plane(...)`